tests/Unit/Domain/Creators/Test_Sphere.cpp

```diff
@@ -59,324 +63,321 @@ create_boundary_condition() {
+
+  MAKE_GENERATOR(gen);
+
+  for (size_t block_id = 0; block_id < num_blocks; ++block_id) {
+    const auto& block = blocks[block_id];
+    const auto& boundary_conditions = all_boundary_conditions[block_id];
+
+    // verify if adjacent to inner block
+    const auto abuts_inner_block =
+        [&num_blocks](const auto& direction_and_neighbor) {
+          const auto& [direction, neighbor] = direction_and_neighbor;
+          (void)direction;
+          if (neighbor.id() == (num_blocks - 1)) {
+            return true;
+          } else {
+            return false;
+          }
+        };
+
+    tnsr::I<double, 3, Frame::BlockLogical> block_logical_coords =
+        logical_coords(gen, num_blocks, block_id,
+                       alg::any_of(block.neighbors(), abuts_inner_block));
+
+    const auto corner_distance_from_origin =
+        [&block, &block_logical_coords, &current_time, &func_of_time,
+         &time_dependencies,
+         &velocity](const auto& initial_expiration_time) -> double {
+      // use stationary map if independent of time
+      if (time_dependencies.size() == 0) {
+        auto location_time_indep =
+            block.stationary_map()(block_logical_coords, 0.0)
+                .get_vector_of_data()
+                .second;
+        return sqrt(square(location_time_indep[0]) +
+                    square(location_time_indep[1]) +
+                    square(location_time_indep[2]));
       } else {
-        local_coord_maps.emplace_back(
-            make_coordinate_map_base<Frame::BlockLogical, TargetFrame>(
-                Affine3D{Affine(-1.0, 1.0, -1.0 * inner_radius / sqrt(3.0),
-                                inner_radius / sqrt(3.0)),
-                         Affine(-1.0, 1.0, -1.0 * inner_radius / sqrt(3.0),
-                                inner_radius / sqrt(3.0)),
-                         Affine(-1.0, 1.0, -1.0 * inner_radius / sqrt(3.0),
-                                inner_radius / sqrt(3.0))}));
+        current_time = initial_expiration_time.begin()->second;
+        // go from logical to grid coords, then grid to inertial coords
+        auto inertial_location_time_dep =
+            block.moving_mesh_grid_to_inertial_map()(
+                block.moving_mesh_logical_to_grid_map()(
+                    block_logical_coords, current_time, func_of_time),
+                current_time, func_of_time);
+
+        // origin in intertial frame (need to shift inertial
+        // coord by velocity * (final_time - initial_time))
+        return sqrt(square(get<0>(inertial_location_time_dep) -
+                           velocity[0] * (current_time - 1.0)) +
+                    square(get<1>(inertial_location_time_dep) -
+                           velocity[1] * (current_time - 1.0)) +
+                    square(get<2>(inertial_location_time_dep) -
+                           velocity[2] * (current_time - 1.0)));
+      }  // end time-dependent if/else
+    };   // end lambda
```

Comment:
I think you can just evaluate the lambda right here instead of calling it later. I think you can also get rid of the `initial_expiration_times` altogether. You can just hardcode a time, like you already do in some places. Best use a consistent `time` constant everywhere.